src/index.ts

```diff
@@ -259,20 +364,163 @@ export class Authenticator {
     return tokens;
   }
 
+  /**
+   * Extract values of the CSRF tokens from the request cookies.
+   * @param  {Array}  cookieHeaders 'Cookie' request headers.
+   * @return {CSRFTokens} Extracted CSRF Tokens from cookie.
+   */
+  _getCSRFTokensFromCookie(cookieHeaders: Array<{ key?: string | undefined, value: string }> | undefined): CSRFTokens {
+    if (!cookieHeaders) {
+      this._logger.debug("Cookies weren't present in the request");
+      throw new Error("Cookies weren't present in the request");
+    }
+
+    this._logger.debug({ msg: 'Extracting CSRF tokens from request cookie', cookieHeaders });
+
+    const cookies = cookieHeaders.flatMap(h => Cookies.parse(h.value));
+    const csrfTokens: CSRFTokens = cookies.reduce((tokens, {name, value}) => {
+      if (name.startsWith(this._cookieBase)) {
+        [
+          NONCE_COOKIE_NAME_SUFFIX,
+          NONCE_HMAC_COOKIE_NAME_SUFFIX,
+          PKCE_COOKIE_NAME_SUFFIX,
+        ].forEach(key => {
+          if (name.endsWith(`.${key}`)) {
+            tokens[key] = value;
+          }
+        });
+      }
+      return tokens;
+    }, {});
+
+    this._logger.debug({ msg: 'Found CSRF tokens in cookie', csrfTokens });
+    return csrfTokens;
+  }
+
+  async _revokeTokens(tokens: Tokens) {
+    const authorization = this._getAuthorization();
+    const revokeRequest = {
+      url: `https://${this._userPoolDomain}/oauth2/revoke`,
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+        ...(authorization && {'Authorization': `Basic ${authorization}`}),
+      },
+      data: stringify({
+        client_id: this._userPoolAppId,
+        token: tokens.refreshToken,
+      }),
+    } as const;
+    this._logger.debug({ msg: 'Revoking refreshToken...', request: revokeRequest, refreshToken: tokens.refreshToken });
+    return axios.request(revokeRequest)
+      .then(() => {
+        this._logger.debug({ msg: 'Revoked refreshToken', refreshToken: tokens.refreshToken });
+      })
+      .catch(err => {
+        this._logger.error({ msg: 'Unable to revoke refreshToken', request: revokeRequest, err: JSON.stringify(err) });
+        throw err;
+      });
+  }
+
+  async _clearCookies(event: CloudFrontRequestEvent, tokens: Tokens = {}): Promise<CloudFrontRequestResult> {
+    this._logger.info({ msg: 'Clearing cookies...', event, tokens });
+    const { request } = event.Records[0].cf;
+    const cfDomain = request.headers.host[0].value;
+    const requestParams = parse(request.querystring);
+    const redirectURI = requestParams.redirect_uri as string;
```

Comment:
Seems like `redirectURI` is always undefined. Therefore, request would fail as Location header has no value. I'd introduce a new property `logoutRedirectUri ` inside `LogoutConfiguration` to make logout uri configurable.

Reply:
Good catch!

`redirectURI` is currently being read from the URL query string. So, anyone requesting to logout would need to call the `/${logoutUri}?redirect_uri=https://domain/` path.

However, based on your suggestion, I'm inclining towards going for the `logoutRedirectUri` config param as the default, with the one in the query string as a backup. The reason being, the other way of calling `_clearCookies` (from inside `handleSignOut`) will not have `LogoutConfiguration` set as it's not needed there. What say?

```ts
// logoutRedirectUri as default; redirect_uri as override
const redirectURI = this._logoutConfiguration.logoutRedirectUri ||
  requestParams.redirect_uri as string; // Needed when invoked from the dedicated handleSignIn handler
```

Reply:
Sounds good!

Reply:
Fixed